### DISCUSSION TITLE
Array Mapping

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [linq2db/linq2db#4633] [Q&A]
I see a lot discussions about array support for PG, but I don't see a way to map a int[] to PG array,.
How to do the mapping? Especially in a code first project, how to specify the data type?

**Comment by HypeillingerChillinger** [comment]
Try this:
  
        entityBuilder.Property(x => x.Statuses)
            .HasConversion(x => x!.Select(y => (int)y).ToArray(), x => x.Select(y => (YourType)y).ToArray())
            .HasColumnName("statuses")
            .HasDbType("int[]");

**Comment by MaceWindu** [comment]
What kind of issue you have? We don't have special support for arrays currently, but basic functionality should work out of box. E.g. see https://github.com/linq2db/linq2db/blob/master/Tests/Linq/DataProvider/PostgreSQLExtensionsTests.cs. There we only specify `DbType` to be able to create test table, as we don't infer database type for arrays automatically
